src/index.d.ts

```diff
@@ -6,7 +6,14 @@ export function props(elem: any, data?: PropData): void;
 
 export const emit: (elem: any, eventName: string, eventOptions = {}) => boolean;
 
-export class Component extends HTMLElement {}
+export class Component extends HTMLElement {
+  updatedCallback?(prevProps: { [name: string]: any }): boolean;
```

Comment:
that is not true @andreawyss, if it was interface and user would use `implements`, then it would errored, not with this one.

By having it on class definition, user is not forced to use that, it just provides better intellisense  within editor.